modules/xfem/src/userobjects/MeshCut2DFractureUserObject.C

```diff
@@ -26,25 +26,43 @@ MeshCut2DFractureUserObject::validParams()
   InputParameters params = MeshCut2DUserObjectBase::validParams();
   params.addClassDescription("XFEM mesh cutter for 2D models that defines cuts with a"
                              "mesh and uses fracture integrals to determine growth");
-  params.addRequiredParam<Real>("k_critical", "Critical fracture toughness.");
-  params.addRequiredParam<Real>("growth_increment", "Length to grow crack if k>k_critical");
-  params.addParam<unsigned int>("ring_number", 1, "Fracture integral ring number");
+  params.addRequiredParam<Real>("growth_increment",
+                                "Length to grow crack if k>k_critical or stress>stress_threshold");
+  params.addParam<Real>("k_critical", "Critical fracture toughness.");
+  params.addParam<Real>("stress_threshold", "Stress threshold for growing crack");
+  params.addParam<VectorPostprocessorName>(
+      "ki_vectorpostprocessor", "II_KI_1", "The name of the vectorpostprocessor that contains KI");
+  params.addParam<VectorPostprocessorName>("kii_vectorpostprocessor",
+                                           "II_KII_1",
+                                           "The name of the vectorpostprocessor that contains KII");
+  params.addParam<VectorPostprocessorName>(
+      "stress_vectorpostprocessor",
+      "The name of the vectorpostprocessor that contains crack front stress");
   return params;
 }
 
 MeshCut2DFractureUserObject::MeshCut2DFractureUserObject(const InputParameters & parameters)
   : MeshCut2DUserObjectBase(parameters),
-    _k_critical(getParam<Real>("k_critical")),
     _growth_increment(getParam<Real>("growth_increment")),
-    _ring_number_string(std::to_string(getParam<unsigned int>("ring_number")))
+    _use_k(isParamValid("k_critical")),
+    _use_stress(isParamValid("stress_threshold")),
+    _k_critical(_use_k ? getParam<Real>("k_critical") : 1.0e12),
```

Comment:
I assume the 1e12 in this line and the one below are "big numbers". We should use `std::numeric_limits<Real>::max()` instead.